src/ring_theory/ideal/operations.lean

```diff
@@ -1174,7 +1209,51 @@ surjective_quot_mk _
 lemma quotient.mkₐ_ker (I : ideal A) : (quotient.mkₐ R I).to_ring_hom.ker = I :=
 ideal.mk_ker
 
-variable {R}
+variables {R} {B : Type*} [comm_ring B] [algebra R B]
+
+lemma quotient.smul (f : A →ₐ[R] B) (r : R) (x : f.to_ring_hom.ker.quotient) :
+  f.to_ring_hom.ker_lift (r • x) = r • f.to_ring_hom.ker_lift x :=
+begin
+  obtain ⟨a, rfl⟩ := quotient.mkₐ_surjective R _ x,
+  rw [← alg_hom.map_smul, quotient.mkₐ_eq_mk, ring_hom.ker_lift_mk],
+  exact f.map_smul _ _
+end
+
+/-- The induced algebras morphism from the quotient by the kernel to the codomain. -/
+def ker_lift_alg (f : A →ₐ[R] B) : f.to_ring_hom.ker.quotient →ₐ[R] B :=
+alg_hom.mk' f.to_ring_hom.ker_lift (λ _ _, quotient.smul f _ _)
+
+@[simp]
+lemma ker_lift_alg_mk (f : A →ₐ[R] B) (a : A) :
+  ker_lift_alg f (quotient.mk f.to_ring_hom.ker a) = f a := rfl
+
+@[simp]
+lemma ker_lift_alg_to_ring_hom (f : A →ₐ[R] B) :
+  (ker_lift_alg f).to_ring_hom = ring_hom.ker_lift f := rfl
+
+/-- The induced algebra morphism from the quotient by the kernel is injective. -/
+lemma ker_lift_alg_injective (f : A →ₐ[R] B) : function.injective (ker_lift_alg f) :=
+ring_hom.ker_lift_injective f
+
+/-- The first isomorphism theorem for agebras, computable version. -/
+def quotient_ker_alg_equiv_of_right_inverse
+  {f : A →ₐ[R] B} {g : B → A} (hf : function.right_inverse g f) :
+  f.to_ring_hom.ker.quotient ≃ₐ[R] B :=
+{ to_fun := ker_lift_alg f,
+  inv_fun := (quotient.mkₐ R f.to_ring_hom.ker) ∘ g,
+  left_inv := begin
+    rintro ⟨a⟩,
+    apply ker_lift_alg_injective,
+    change f (g (f a)) = f a,
+    exact hf (f a),
+  end,
+  right_inv := hf,
+  ..ker_lift_alg f}
+
+/-- The first isomorphism theorem for agebras. -/
+noncomputable def quotient_ker_alg_equiv_of_surjective {f : A →ₐ[R] B} (hf : function.surjective f) :
+  f.to_ring_hom.ker.quotient ≃ₐ[R] B :=
+quotient_ker_alg_equiv_of_right_inverse (classical.some_spec hf.has_right_inverse)
```

Comment:
Same remark as above: should we add `simp` lemmas?

Reply:
I've added the lemmas. They are called `quotient_ker_equiv_of_right_inverse.apply` and so on. Feel free to suggest better names...